packages/drivers/odsp-driver/src/test/localOdspDriver.spec.ts

```diff
@@ -0,0 +1,240 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import fs from "fs";
+import { strict as assert } from "assert";
+import { IStream } from "@fluidframework/driver-definitions";
+import { IOdspResolvedUrl } from "@fluidframework/odsp-driver-definitions";
+import {
+    IClient,
+    ISequencedDocumentMessage,
+    MessageType,
+    SummaryType,
+} from "@fluidframework/protocol-definitions";
+import { MockLogger } from "@fluidframework/telemetry-utils";
+import { LocalOdspDocumentService } from "../odspDocumentService";
+import { LocalOdspDocumentServiceFactory } from "../odspDocumentServiceFactory";
+import { LocalOdspDocumentStorageService } from "../odspDocumentStorageManager";
+
+describe("Local Odsp driver", () => {
+    // TODO: add end-to-end test
+
+    const fakeOdspResolvedUrl: IOdspResolvedUrl = {
+        type: "fluid",
+        odspResolvedUrl: true,
+        id: "1",
+        siteUrl: "fakeUrl",
+        driveId: "1",
+        itemId: "1",
+        url: "fakeUrl",
+        hashedDocumentId: "1",
+        endpoints: {
+            snapshotStorageUrl: "fakeUrl",
+            attachmentPOSTStorageUrl: "fakeUrl",
+            attachmentGETStorageUrl: "fakeUrl",
+            deltaStorageUrl: "fakeUrl",
+        },
+        tokens: {},
+        fileName: "fakeName",
+        summarizer: false,
+        fileVersion: "1",
+    };
+
+    const localSnapshot = fs.readFileSync(
+        `${__dirname}/../../src/test/localSnapshots/localSnapshot1.json`,
+        { encoding: "utf8" },
+    );
+
+    async function assertThrowsAsync(fn: () => Promise<any>) {
```

Comment:
assert.throws(await fn()); ?
Not sure why you need a wrapper for that.

Reply:
`assert.throws` doesn't play nicely with `async` functions. I initially tried using it but couldn't find a working implementation.
I just had another look into other solutions and found `assert.rejects` which we use in other places in code.

Reply:
assert.rejects() sounds good :)